File: software/examples/javascript/ExampleConfiguration.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'XYZ'; // Change XYZ to the UID of your Performance DC Bricklet

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var pdc = new Tinkerforge.BrickletPerformanceDC(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function (error) {
        console.log('Error: ' + error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function (connectReason) {
        pdc.setDriveMode(Tinkerforge.BrickletPerformanceDC.DRIVE_MODE_DRIVE_COAST);
        pdc.setPWMFrequency(10000); // Use PWM frequency of 10 kHz
        pdc.setMotion(4096, 4096); // Slow ac-/deceleration (12.5 %/s)
        pdc.setVelocity(32767); // Full speed forward (100 %)
        pdc.setEnabled(true); // Enable motor power
    }
);

console.log('Press key to exit');
process.stdin.on('data',
    function (data) {
        // Stop motor before disabling motor power
        pdc.setMotion(4096, 16384); // Fast decceleration (50 %/s) for stopping
        pdc.setVelocity(0); // Request motor stop

        setTimeout(function () {
            pdc.setEnabled(false); // Disable motor power

            ipcon.disconnect();
            process.exit(0);
        }, 2000); // Wait for motor to actually stop: velocity (100 %) / decceleration (50 %/s) = 2 s
    }
);

File: software/examples/rust/example_configuration.rs
use std::{error::Error, io, thread, time::Duration};
use tinkerforge::{ip_connection::IpConnection, performance_dc_bricklet::*};

const HOST: &str = "localhost";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your Performance DC Bricklet.

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection.
    let pdc = PerformanceDcBricklet::new(UID, &ipcon); // Create device object.

    ipcon.connect((HOST, PORT)).recv()??; // Connect to brickd.
                                          // Don't use device before ipcon is connected.

    pdc.set_drive_mode(PERFORMANCE_DC_BRICKLET_DRIVE_MODE_DRIVE_COAST).recv()?;
    pdc.set_pwm_frequency(10000).recv()?; // Use PWM frequency of 10 kHz
    pdc.set_motion(4096, 4096).recv()?; // Slow ac-/deceleration (12.5 %/s)
    pdc.set_velocity(32767).recv()?; // Full speed forward (100 %)
    pdc.set_enabled(true).recv()?; // Enable motor power

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;

    // Stop motor before disabling motor power
    pdc.set_motion(4096, 16384).recv()?; // Fast decceleration (50 %/s) for stopping
    pdc.set_velocity(0).recv()?; // Request motor stop
    thread::sleep(Duration::from_millis(2000)); // Wait for motor to actually stop: velocity (100 %) / decceleration (50 %/s) = 2 s
    pdc.set_enabled(false).recv()?; // Disable motor power

    ipcon.disconnect();
    Ok(())
}
